Task1/WordFrequencyCounter.java

```diff
@@ -0,0 +1,118 @@
+                "Mm, I'm gonna try with a little help from my friends";
+
+        // Remove commas, newline characters, and convert to lowercase
+        song = song.replaceAll("[,\\n]", "").toLowerCase();
+
+        // Split the song into an array of words
+        String[] wordsArray = song.split(" ");
+
+        // Create an array to store word frequencies
+        String[] uniqueWordsArray = new String[wordsArray.length];
+        int[] wordFrequencies = new int[wordsArray.length];
+
+        // Count word frequencies and print the results
+        int uniqueWordCount = 0;
+        for (String word : wordsArray) {
+            int index = indexOfWord(uniqueWordsArray, word, uniqueWordCount);
+            if (index == -1) {
+                uniqueWordsArray[uniqueWordCount] = word;
+                wordFrequencies[uniqueWordCount] = 1;
+                uniqueWordCount++;
+            } else {
+                wordFrequencies[index]++;
+            }
+        }
+
+        // Print unique words and their frequencies
+        for (int i = 0; i < uniqueWordCount; i++) {
+            System.out.println(uniqueWordsArray[i] + ": " + wordFrequencies[i]);
+        }
+
+        // Part 2
+        List<String> wordsList = new ArrayList<>();
+        Set<String> uniqueWordsSet = new HashSet<>();
+
+        // Update Part 1 to use a List to count words
+        for (String word : wordsArray) {
+            if (!uniqueWordsSet.contains(word)) {
+                wordsList.add(word);
+                uniqueWordsSet.add(word);
+            }
+        }
+
+        // Print unique words without duplicates
+        System.out.println("Unique Words: " + wordsList);
+
+        // Sort words by string length
+        Collections.sort(wordsList, (a, b) -> a.length() - b.length());
+        System.out.println("Words sorted by length: " + wordsList);
+
+        // Part 3
+        // Remove all occurrences of "yellow" and "submarine" from the List
+        wordsList.removeIf(word -> word.equals("yellow") || word.equals("submarine"));
+
+        // Print the resulting List
+        System.out.println("List after removing 'yellow' and 'submarine': " + wordsList);
+
+        // Part 4
+        try (FileWriter fileWriter = new FileWriter("song.txt")) {
+            // Save song "beatles yellow submarine" to a txt file
+            fileWriter.write("beatles yellow submarine");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        try (FileReader fileReader = new FileReader("song.txt");
+             BufferedReader bufferedReader = new BufferedReader(fileReader)) {
+            String line;
+            StringBuilder songFromFile = new StringBuilder();
+            while ((line = bufferedReader.readLine()) != null) {
+                songFromFile.append(line);
+            }
+            // Define a custom exception: BeatlesException
+            try {
+                // Create a method to verify if a random string is found in the song
```

Comment:
right now, the logic to check if a specific string is in the song is mixed with file reading code. Extract this logic into a new method named, for ex: verifyStringInSong. This method will take the song’s content and the string you’re searching for as parameters. If the string is not found, it throws the custom BeatlesException.